src/Storefront/Theme/ScssPhpCompiler.php

```diff
@@ -19,21 +21,46 @@ class ScssPhpCompiler extends AbstractScssCompiler
      */
     private readonly ?array $cacheOptions;
 
+    private ?TagAwareCacheInterface $cache = null;
+
     /**
      * @param array<string, mixed>|null $cacheOptions
      */
-    public function __construct(?array $cacheOptions = null)
+    public function __construct(?array $cacheOptions = null, ?TagAwareCacheInterface $cache = null)
     {
-        $this->compiler = new Compiler($cacheOptions);
+        $this->compiler = new Compiler();
         $this->cacheOptions = $cacheOptions;
+        $this->cache = $cache;
     }
 
     public function reset(): void
     {
-        $this->compiler = new Compiler($this->cacheOptions);
+        $this->compiler = new Compiler();
     }
 
     public function compileString(AbstractCompilerConfiguration $config, string $scss, ?string $path = null): string
+    {
+        if ($config->getValue('skipCache') === true || !$this->cacheOptions || !$this->cache) {
+            return $this->doCompileString($config, $scss, $path);
+        }
+
+        $cacheKey = $this->generateCacheKey($config, $scss, $path);
+        $cachedResult = $this->cache->get($cacheKey, function ($item) use ($config, $scss, $path) {
+            if (isset($this->cacheOptions['lifetime'])) {
```

Comment:
do we need to check, if `lifetime` has a valid value? like is an integer greater than 0 or so?